code/modules/mob/living/basic/bots/dedbot.dm

```diff
@@ -0,0 +1,117 @@
+	gold_core_spawnable = HOSTILE_SPAWN
+	limb_destroyer = TRUE
+	bubble_icon = "machine"
+	pass_flags = PASSMOB | PASSFLAPS
+	maximum_survivable_temperature = 360 //prone to overheating
+	possessed_message = "You are an exenteration drone. Exenterate."
+	additional_access = /datum/id_trim/away/hauntedtradingpost/boss
+	bot_mode_flags = BOT_MODE_ON | BOT_MODE_AUTOPATROL
+	mob_size = MOB_SIZE_SMALL
+	robot_arm = /obj/item/hatchet/cutterblade
+	density = FALSE
+	COOLDOWN_DECLARE(trigger_cooldown)
+	//time between exenteration uses
+	var/exenteration_cooldown_duration = 0.5 SECONDS
+	//aoe slash ability
+	var/datum/action/cooldown/mob_cooldown/bot/exenterate
+
+/mob/living/basic/bot/dedbot/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/death_drops, /obj/effect/gibspawner/robot)
+	var/static/list/innate_actions = list(
+	SPIN_SLASH_ABILITY_TYPEPATH = BB_DEDBOT_SLASH,
+	)
+	grant_actions_by_list(innate_actions)
+
+/datum/ai_controller/basic_controller/bot/dedbot
+	max_target_distance = 1
+	blackboard = list(
+		BB_TARGETING_STRATEGY = /datum/targeting_strategy/basic,
+	)
+	ai_movement = /datum/ai_movement/jps/bot
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/targeted_mob_ability/exenterate,
+		/datum/ai_planning_subtree/respond_to_summon,
+		/datum/ai_planning_subtree/find_patrol_beacon,
+		/datum/ai_planning_subtree/manage_unreachable_list,
+	)
+	max_target_distance = AI_BOT_PATH_LENGTH
+	///keys to be reset when the bot is reseted
+	reset_keys = list(
+		BB_BEACON_TARGET,
+		BB_PREVIOUS_BEACON_TARGET,
+		BB_BOT_SUMMON_TARGET,
+	)
+
+/datum/ai_planning_subtree/targeted_mob_ability/exenterate
+	ability_key = BB_DEDBOT_SLASH
+	finish_planning = FALSE
+
+/datum/action/cooldown/mob_cooldown/exenterate
+	name = "Exenterate"
+	desc = "Disembowel every living thing in range with your blades."
+	button_icon = 'icons/obj/weapons/stabby.dmi'
+	button_icon_state = "huntingknife"
+	click_to_activate = FALSE
+	background_icon = 'icons/hud/guardian.dmi'
+	background_icon_state = "base"
+	cooldown_time = 0.5 SECONDS
+	// radius in tiles of AOE effect
+	var/ability_range = 1
+	// how much damage this ability does
+	var/damage_dealt = 18
+	// factions we dont attack
+	var/immune_factions = list(ROLE_SYNDICATE)
+	// weighted list of body zones this can hit
+	var/static/list/valid_targets = list(
+		BODY_ZONE_CHEST = 2,
+		BODY_ZONE_R_ARM = 1,
+		BODY_ZONE_L_ARM = 1,
+		BODY_ZONE_R_LEG = 1,
+		BODY_ZONE_L_LEG = 1,
+	)
+
+/datum/action/cooldown/mob_cooldown/exenterate/Activate(atom/caster)
+	if(!COOLDOWN_FINISHED(src, cooldown_time))
+		return FALSE
+	caster.Shake(1.4, 0.8, 0.3 SECONDS)
+	caster.visible_message(span_danger("[caster] shakes violently!"))
+	playsound(caster, 'sound/weapons/drill.ogg', 120 , TRUE)
+	slash_em(caster)
+	StartCooldown(cooldown_time)
+
+/datum/action/cooldown/mob_cooldown/exenterate/proc/slash_em(atom/caster, mob/living/victim)
+	for(victim in range(ability_range, caster))
```

Comment:
```suggestion
/datum/action/cooldown/mob_cooldown/exenterate/proc/slash_em(atom/caster)
	for(var/mob/living/victim in range(ability_range, caster))
```
You could also add a can_reach() check to spare unreachable mobs from the mayehm.